class.jetpack-gutenberg.php

```diff
@@ -459,20 +458,26 @@ public static function load_styles_as_required( $type ) {
 	 * Only enqueue block scripts when needed.
 	 *
 	 * @param string $type Slug of the block.
-	 * @param array  $script_dependencies An array of view-side Javascript dependencies to be enqueued.
 	 *
 	 * @since 7.2.0
 	 *
 	 * @return void
 	 */
-	public static function load_scripts_as_required( $type, $script_dependencies = array() ) {
+	public static function load_scripts_as_required( $type ) {
```

Comment:
I think we'll still need to keep `$script_dependencies` parameter around to allow adding additional dependencies that cannot be imported, such as here `thickbox`:

https://github.com/Automattic/jetpack/pull/9802/files#diff-4e9241dbd5a466cb517f9e815001295aR201

...or anything else that's not dealt with this list:

https://github.com/Automattic/wp-calypso/blob/2640ee82bf2b0d390d7ba90afb9a2332c4e322cf/packages/wordpress-external-dependencies-plugin/index.js#L20-L46

See the full list https://developer.wordpress.org/reference/functions/wp_enqueue_script/#default-scripts-included-and-registered-by-wordpress

Reply:
My initial reaction is that I still prefer managing dependencies in the script that use them.

There's no reason that an import must correspond to a published npm package, we can add rules to the webpack plugin to handle more of these cases. Then we'd `import 'thickbox';` where we depend on the script and continue to leverage the tooling.

Reply:
Yup, that makes sense!

I can imagine three options;
- `thickbox` and other WP deps added to the list in `@automattic/wordpress-external-dependencies-plugin`
- `@automattic/wordpress-external-dependencies-plugin` can take additional list of externals from the consumer (Jetpack)
- We define these as externals in Jetpack's Webpack config (tho we'd still have to define them outside the `*.deps.json` so I suppose this isn't the way to go.

Anything I don't see and which way would you prefer?

Reply:
> * `@automattic/wordpress-external-dependencies-plugin` can take additional list of externals from the consumer (Jetpack)

I was thinking about something similar in the context of plugins which can define their own shared scripts. It would be useful if you want to use some popular libraries.

Reply:
I was literally writing the same comment. :-) 

We might have e.g. 10 blocks that all depend on same external non-wp library. These blocks are likely ending up on the same page so we wouldn't want to bundle them together or have them async load the library dynamically so that the site's caching/concatenation plugins can handle it.

Totally optimizing for future so just food for thought.

Reply:
Totally in line with what folks shared on https://make.wordpress.org/meta/2019/03/08/the-block-directory-and-a-new-type-of-plugin/. I personally would love to have this plugin optimized for Block Directory.

Reply:
It may make sense to include some more core scripts like `thickbox`. In the long term, the most value will come from this:

> * `@automattic/wordpress-external-dependencies-plugin` can take additional list of externals from the consumer (Jetpack)

I was planning to add this feature when necessary, seems like now is the time 🙂 

Tracking: https://github.com/Automattic/wp-calypso/issues/32097

There was already some interest during initial implementation: 6-gh-woocommerce-payments